data/binding/listbinding.go

```diff
@@ -1,41 +1,45 @@
 package binding
 
+import "errors"
+
+var errOutOfBounds = errors.New("index out of bounds")
+
 // DataList is the base interface for all bindable data lists.
 //
 // Since: 2.0.0
 type DataList interface {
 	DataItem
-	GetItem(int) DataItem
+	GetItem(int) (DataItem, bool)
 	Length() int
 }
 
 type listBase struct {
 	base
-	val []DataItem
+	items []DataItem
 }
 
 // GetItem returns the DataItem at the specified index.
-func (b *listBase) GetItem(i int) DataItem {
-	if i < 0 || i >= len(b.val) {
-		return nil
+func (b *listBase) GetItem(i int) (DataItem, bool) {
```

Comment:
I wonder if `GetItem(i int) (DataItem, error)` would be better so it can return `errOutOfBounds`. Although this means DataList would deviate from DataMap, it seems the least-surprising approach

Reply:
It depends if we want to match the primitive types behaviour or a consistent higher level API design.
As you say currently Get/Set return an error, whereas GetItem for list and map are both bools.
The only error GetItem could return is the out of bounds, so is a generic error that then has to be checked for the reason the most helpful approach?

Reply:
Yes, the only error GetItem could return from these in-memory list bindings is out-of-bounds. But if someone implemented a list binding to the lines in a file, or the rows in a database, they could generate a plethora of errors.

Reply:
Are you sure? This is the lookup that returns a binding to a value - so it doesn't do anything with the values...
I think you could make the same argument that map lookup should be `error` instead of `bool` - so I still think they should be consistent, whichever we use.